### PR TITLE
add schema for cleaner OAI-PMH-output

### PR DESCRIPTION
**Pull request by martinobersteiner**
before this commit
  - all fields in LOM-jsons were output to OAI-PMH
  - the order of output wasn't guranteed

however, harvesters may depend on order and/or absence of extra fields
hence, this commit adds a `LOMToOAISchema` in order to both,
  - order OAI-PMH output as per LOM-spec
  - exclude extra fields

this new schema made the `resources/serializers/schema.py` too large
hence, it's refactored into a subpackage

### CODE REVIEW

File: invenio_records_lom/cli.py [hunk 1]
```diff
@@ -33,6 +34,32 @@ def rebuild_index():
...
 
 
+@lom.command()
+@with_appcontext
+def check():
+    """Check records in SQL-database against marshmallow-schema for OAI.
+
+    Note: this does not guarantee by itself that OAI-PMH API works correctly, since
+    (1) Records in opensearch might not mirror records in SQL-database, call `invenio lom reindex` to remedy this.
+    (2) Records that pass OAI-schema verification might still not play nicely with OER-Hub
```
Comment (utnapischtim) [inline invenio_records_lom/cli.py:44]: don't mention OERhub directly. use the general word `harvesters`.

File: invenio_records_lom/cli.py [hunk 2]
```diff
@@ -33,6 +34,32 @@ def rebuild_index():
     click.secho("Reindexed records!", fg="green")
 
 
+@lom.command()
+@with_appcontext
+def check():
```
Comment (utnapischtim) [inline invenio_records_lom/cli.py:39]: minor: normally i would add an `click.option` to check a specific `pid` too. you can keep the loop, but what you put into the loop depends on if a specific check is requested or not.

File: invenio_records_lom/resources/serializers/__init__.py [hunk 3]
```diff
@@ -106,13 +121,15 @@ def build_location(self, jsn, parent_tag):
         """Append location to parent_tag."""
         tag = self.element_maker.location(
             jsn["#text"],
-            **{"{http://www.w3.org/XML/1998/namespace}lang": jsn["type"]},
+            # while some of LOM-UIBK's examples include a type-attribute here, most don't
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/__init__.py:124]: if you talk about the LOM-UIBK here, then you should mention it somewhere with a link. maybe in the README?

File: invenio_records_lom/resources/serializers/schemas/ui.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                "icon": icon,
+            }
+
+        return [right]
+
+
+class LOMUIBaseSchema(BaseObjectSchema):
+    """Base schema for LOMUI-classes, containing all common fields."""
+
+    # publication_date_l10n_medium = FormatEDTF(...)
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/schemas/ui.py:106]: is this code really necessary to keep it commented? if yes, please add a comment why it should stay. (applies to the whole commented block)

**Review by utnapischtim — COMMENTED** [review]
Improve the commit message by adding arguments why the refactoring was necessary and improves the code. Add further points who describes the bug fixing. the best would be to separate refactoring and bug fixing in different commits, but that is normally difficult.

File: invenio_records_lom/cli.py (hunk 1 above)
Comment (martinobersteiner) [reply invenio_records_lom/cli.py:44]: done

File: invenio_records_lom/cli.py (hunk 2 above)
Comment (martinobersteiner) [reply invenio_records_lom/cli.py:39]: wanted to eventually add one too
no time like today, I guess
added

File: invenio_records_lom/resources/serializers/schemas/ui.py (hunk 4 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas/ui.py:106]: whoops, definitely not necessary
this was for debugging landing page in last commit (it's all the fields that the comparable `invenio` template has)
removed

File: invenio_records_lom/resources/serializers/__init__.py (hunk 3 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/__init__.py:124]: done

**Pull request merged**